src/Tags/TagWatcher.lua

```diff
@@ -13,7 +14,12 @@ function TagWatcher.new(fabric)
 
 	fabric:on("componentRegistered", function(staticComponent)
 		if staticComponent.tag then
-			self:listenForTag(staticComponent.tag, staticComponent)
+			local connection
+			connection = HeartbeatWrapper.Heartbeat:Connect(function()
+				connection:Disconnect()
+
+				self:listenForTag(staticComponent.tag, staticComponent)
+			end)
```

Comment:
See how this is done in `_deferCreation`. We should implement it like this, with a queue. Not a new event listener for every tag.